### PR TITLE
[pr-test]add basic httpsvr+pipeline test

### PR DESCRIPTION
**Pull request by benja-wu**
* This script is for **Jekins**'s GitHub hook execution command. 
* It will start up an Easegress instance, create HTTPServer + Pipeline, and run a backend, then curl the HTTPServer at last.
* The script's final execution result depends on the curl HTTP response code.

### CODE REVIEW

File: build/test/httpserver_test.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+# an HTTPServer and Pipeline for testing HTTP Requests.
+# author:   benjaminwu 
+# date:     2021/0716
+
+# path related define.
+SCRIPTPATH=`pwd -P`
+pushd $SCRIPTPATH"/../../example" > /dev/null
+EXAMPDIR=`pwd -P`
+WRITER01DIR=$EXAMPDIR"/writer-001"
+
```
Comment (zhao-kun) [inline build/test/httpserver_test.sh:12]: why not `set -e` ?

File: build/test/httpserver_test.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+# Test the Easegress' basic functionality which is generating
+# an HTTPServer and Pipeline for testing HTTP Requests.
+# author:   benjaminwu 
+# date:     2021/0716
+
+# path related define.
+SCRIPTPATH=`pwd -P`
+pushd $SCRIPTPATH"/../../example" > /dev/null
+EXAMPDIR=`pwd -P`
+WRITER01DIR=$EXAMPDIR"/writer-001"
```
Comment (zhao-kun) [inline build/test/httpserver_test.sh:11]: `SCRIPTPATH=$(dirname $(realpath ${BASH_SOURCE[0]}))`

`pwd -P` isn't always the script's path.

If we locate script path, the example path should be got by relative path to scriptpath

Comment (benja-wu) [reply build/test/httpserver_test.sh]: Updated as comment.

File: build/test/httpserver_test.sh [hunk 3]
```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+# Test the Easegress' basic functionality which is generating
+# an HTTPServer and Pipeline for testing HTTP Requests.
+# author:   benjaminwu 
+# date:     2021/0716
```
Comment (haoel) [inline build/test/httpserver_test.sh:5]: Let's remove the author and date and put Apache 2.0 License in the head of this file.

File: build/test/httpserver_test.sh [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+# clean the cluster resource first.
+clean
+
+# start writer01 for testing. 
+start_svr=`$WRITER01DIR/start.sh `
+
+# wait Easegress to be ready
+sleep 2
```
Comment (haoel) [inline build/test/httpserver_test.sh:57]: can we use a better way to do this? such as https://unix.stackexchange.com/questions/5277/how-do-i-tell-a-script-to-wait-for-a-process-to-start-accepting-requests-on-a-po

File: build/test/httpserver_test.sh [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    kind: Proxy
+    mainPool:
+      servers:
+      - url: http://127.0.0.1:9095
+      loadBalance:
+        policy: roundRobin' | $WRITER01DIR/egctl.sh object create
+
+# run the backend.
+(go run $backend & )
+sleep 2
```
Comment (haoel) [inline build/test/httpserver_test.sh:96]: Ditto

File: build/test/httpserver_test.sh (hunk 3 above)
Comment (benja-wu) [reply build/test/httpserver_test.sh]: Updated.

File: build/test/httpserver_test.sh (hunk 4 above)
Comment (benja-wu) [reply build/test/httpserver_test.sh]: Replacing it with `while ! nc -z localhost $mirror_port</dev/null; do sleep 3; done`.

**Review by haoel — APPROVED** [review]

**Review by localvar — APPROVED** [review]

File: build/test/httpserver_test.sh [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    clean $mirror_pid
+    exit 4
+else 
+    echo -e "\n${COLOR_INFO}test succ${COLOR_NONE}"
+fi
+
+# clean all created resources.
+clean $mirror_pid
+
+popd > /dev/null
```
Comment (xxx7xxxx) [inline build/test/httpserver_test.sh:139]:
```suggestion
popd > /dev/null

```

**Review by xxx7xxxx — APPROVED** [review]

**Pull request merged**